Display date of last update on each page

renestalder opened the pull request:
Add the output of `changed` to the base layout to display
the date of change of the content.

- [x] Add core function
- [x] Format date into a user friendly format
- [x] Styling
- [x] Recommended: Do not show the date for the start page, as it can be misinterpreted as "Oh, there was no change in the whole guide since X".

closes #159

Commit:
feat: render `changed` in layout

Add the output of `changed` to the base layout to display
the date of change of the content.

Commit:
feat: add formatDate Handlebars helper

Format the date of the last change in a user friendly format
when rendered on the page.

Commit:
fix: style meta-info component

gurix commented on src/templates/layout.hbs:
```diff
@@ -51,8 +51,12 @@
         </div>
-
         {{> "content/prev_next/prev_next" prev=previousPage next=nextPage}}
+        {{#if changed}}
```
I like the idea of having a general part that holds meta informations. But then we have to toggle the display in `src/components/content/meta-info/meta-info.hbs`.

renestalder replied:
@gurix I can move it into the component.

gurix commented:
There are already some libraries like https://www.npmjs.com/package/dateformat to format the date in a user friendly name. Is there any reason why we are not simply use this?

renestalder commented:
> There are already some libraries like https://www.npmjs.com/package/dateformat to format the date in a user friendly name. Is there any reason why we are not simply use this?

@gurix Because that's literally done in a couple of vanilla JavaScript lines, so no need to add an additional dependency to the project as long as there is no heavy date formatting stuff going on.

Commit:
fix: do not show meta information for homepage

Hide meta information on homepage to prevent confusion about for what
part of the guide the `changed` date is meant for.

renestalder commented:
@gurix This is now ready for a review.

I wonder if the meta data section needs a visually hidden title or not. Currently it doesn't have one.

gurix commented:
@renestalder In my honest opinion it would need a title because of [WCAG 2.4.6](https://www.w3.org/TR/2008/REC-WCAG20-20081211/#navigation-mechanisms-descriptive) But it is a good question because the only information is the date when the page was changed, nothing more.  I asked myself also if a definition list with only one item is appropriate.  What are you thinking? And can you move the if into the component.

renestalder commented:
> I asked myself also if a definition list with only one item is appropriate. What are you thinking?

You're right. A list with one item is a stupid idea. I'll change that. Including the move of the condition into the component.

renestalder commented:
@gurix I see a logical problem with moving the if condition. Currently the inclusion of the meta component is wrapped inside the aside with a layout CSS class. It totally makes sense to have this part in the layout and not in the component. Moving the if condition to the component makes the layout wrapper `aside` always rendering OR the aside with the layout styles has to be moved into the component too, which is kind of mentally breaking the separation between layout and components. 

I feel it still makes sense to have the general if condition, whether the component should render or not, on the layout level. Moving everything into the component makes the component not usable in other contexts. At least my approach for components is always to make them as isolated enough to reuse them.

Commit:
fix: add visually hidden title for the page's meta information

renestalder commented:
Changed markup and added hidden title.
That's what it looks like currently. `About this page`.

![screenshot 2018-12-07 at 16 08 18](https://user-images.githubusercontent.com/105358/49655245-71da9e00-fa3a-11e8-9688-9a39eac88e24.png)

gurix commented:
In my opinion everything is good expect that the condition to check the date should be in the component and not in the layout. Would you like to adjust this or can I fork `NothingAG:feat/public-change-date` somehow and change it for my own?

renestalder commented:
> In my opinion everything is good expect that the condition to check the date should be in the component and not in the layout.

What do you say about the remarks I mentioned earlier? It's quite common having the check outside the component in UI frameworks as this allows you to only load the component assets when the condition appears true. I know this doesn't apply to this situation here, but from my experience, the way it is currently implemented is the way to go. I can move it into the component, but it seems wrong to me.

gurix commented:
> What do you say about the remarks I mentioned earlier?

Sorry I missed your comment, my bad. I see the your point and yes it would be not good to have an empty aside. I always thought meta info is something like a component that can be used in different templates but it is only used in the layout. From this point of view it makes sense to have the condition there. The question is, what do we do in the future, when more meta infos are needed and they are not all present on every page. But we could discuss this when we reach this point.

I did not attend the last hackday, how is the review process right now? Should I merge it?

renestalder commented:
> The question is, what do we do in the future, when more meta infos are needed and they are not all present on every page. But we could discuss this when we reach this point.

That's a valid point. The condition then would be that the required fields/data (could be "changed date" only for example or multiple other fields) are present an then I see that the encapsulation into the component would probably work better.

> I did not attend the last hackday, how is the review process right now? Should I merge it?

Rule of thumb probably is to let it validate by A4A if it's a fundamental content change or functional change. For other changes that do not change functionality or e.g. only fix typos in texts, I merge directly or ask for a one person review from someone else in this project.

For this issue here, I think we should quickly ask A4A to make an accessibility check.

gurix commented:
Alright, I sent @AndreasUe an email and ask for a quick review.

Commit:
Merge branch 'origin/releases/1.1' into feat/public-change-date

jmuheim commented:
Nice one.

One minor thing: shouldn't it be "Last update: XXX" instead of "Last updated: XXX"?

backflip commented:
@jmuheim, I have opened https://github.com/Access4all/adg/issues/248

jmuheim commented:
Thank you, @backflip!
